### ISSUE TITLE
Golang source sets (go_source) are not supported: source files are unsynced

### ISSUE THREAD

**Issue by wingsofovnia** [bazelbuild/intellij#1265]
Bazel golang rule supports [source sets](https://github.com/bazelbuild/rules_go/blob/master/go/core.rst#go-source) and their further [embedding](https://github.com/bazelbuild/rules_go/blob/master/go/core.rst#embedding) into binaries/tests (https://github.com/bazelbuild/rules_go/blob/master/go/core.rst#embedding). 

We are using this extensively and unfortunately, this is not supported by Bazel Intellij plugin: all sources defined via `go_source` & `embed` attr are "unsynced".

### .bazelproject
```
directories:
  .
derive_targets_from_directories: false
targets:
  //myapp:all
additional_languages:
  go
```
### Bazel build file
```
go_source(
    name = "src",
    srcs = glob(
        include = ["*.go"],
        exclude = ["*_test.go"],
    ),
    deps = [],
)
go_binary(
    name = "app",
    embed = [":src"],
)
```

## Expected behavior
All source files matched by `go_source#srcs` are synced.

## Actual behavior
`go_binary#embed` is ignored and all sources are unsynced.

## Walkaround
It seems that Bazel plugin doesn't understand `go_source` so all sources must be specified explicitly in `go_binary#srcs`. This syncs up correctly:
### Bazel build file
```
go_binary(
    name = "app",
    srcs = glob(
        include = ["*.go"],
        exclude = ["*_test.go"],
    ),
)
```

**Comment by chaoren** [comment]
Is a go_source just a go_library without the import path?

**Comment by wingsofovnia** [comment]
@chaoren I am not sure I understand the question. We use `go_source` for grouping sources so that they can be reused in `go_binary` and `go_test`. It might be `go_library` instead of `go_binary` in my example.

**Comment by wingsofovnia** [comment]
@chaoren I've just discovered that `go_test` doesn't experience the same problem! Giving that Bazel build file looks the following way:
```
go_source(
    name = "src",
    srcs = glob(
        include = ["*.go"],
        exclude = ["*_test.go"],
    ),
    deps = [],
)
go_binary(
    name = "app",
    embed = [":src"],
)
go_test(
    name = "test",
    srcs = glob(
        include = ["*_test.go"],
    ),
    embed = [":src"],
    deps = [
        "@com_github_stretchr_testify//assert:go_default_library",
    ],
)
```

`.bazelproject` with `targets: //myapp:all` will leave all `"*.go"` unsynced, **however**, `"*_test.go"` from `go_test` will be synced!

**Comment by chaoren** [comment]
Yeah, only embeds with tests are handled as that's the only kind of embedding allowed internally. The various kinds of embedding in the bazel rule complicates the situation a bit, as it now allows multiple distinct rules to occupy the same go package.

**Comment by wingsofovnia** [comment]
@chaoren what would be the best way to write similar https://github.com/bazelbuild/intellij/issues/1265#issuecomment-542553601 Bazel build file avoiding src definition duplication? Should I just define  `go_binary` explicitly and embed it to the test?

**Comment by chaoren** [comment]
> Should I just define `go_binary` explicitly and embed it to the test?

That would work yes.

**Comment by wingsofovnia** [comment]
@chaoren Thank you. Do you plan to support embeds with binaries/libraries in the future tho?

**Comment by wingsofovnia** [comment]
@chaoren embedding `go_binary` to `go_test` is deprecated and it is not possible to embed it into `genrule`. In a real project, source and dependency lists are quite long and duplicating it for every rule makes it easier to forget to change something and adds a lot of boilerplate code, therefore, I believe it is not a feasible solution. In other words, for big projects with long BUILD files, this plugin becomes unusable :(

**Comment by sgowroji** [comment]
Hi there! We're doing a clean up of old issues and will be closing this one. Please reopen if you’d like to discuss anything further. We’ll respond as soon as we have the bandwidth/resources to do so.

**Comment by lbcjbb** [comment]
Hi,

I hope this issue can be reopened. In my project I use a lot of `go_source` rules for all generated mock with [mockey](https://github.com/vektra/mockery).

A few months ago I created a simple Bazel project to test all possibiliies to use a `go_source` in a `go_library` and `go_test` rule: https://github.com/leboncoin/bazel-playground/tree/intellij-go-source.

I've just submitted the PR #5205 to support the use of an `go_source` rule in a `go_test` or `go_library` rule, and these submitted changes fix my plugin in Intellij UE. All my generated mock files are synced to the project.
